### PR TITLE
add make upgrade for oep-18 compliance

### PR DESCRIPTION
**Pull request by estute**
I wanted to install code-annotations here, but realized it was not compliant w/ oep-18. Before installing it, I want to first make sure that tests pass w/ oep-18 compliance, then will do the installation in a subsequent PR.

### CODE REVIEW

File: requirements/dev.in [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Packages required for local development
--r test.txt
--r docs.txt
+-r test.in
+-r docs.in
```
Comment (jmbowman) [inline requirements/dev.in]: These should still be the `*.txt` files, to ensure that the same versions are used across development, tests, and production.

Comment (jmbowman) [inline requirements/dev.in]: Also, it seems like this should include `pip-tools.txt` as well.

File: requirements/production.in [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Packages required in a production environment
+-r base.in
```
Comment (jmbowman) [inline requirements/production.in]: Again, should be `base.txt`.

**Comment by estute** [comment]
In order to get this working, I needed to make a couple of fixes:
* I am NOT including requirements/doc.txt in requirements/test.in. This was causing a lot of failures that can be sorted out later.
* I pinned pyyaml to the production value. Some other library was trying to install a different version, and I did not want to include a production environment change in this work.
* various pinning/unpinning.

**Comment by estute** [comment]
^ @bbaker6225 @zubair-arbi @mikix would any of you be able to review this? I picked you are the top contributors to this repo.

File: requirements/dev.in [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+-r test.txt
+
+django-debug-toolbar==1.6
+
+# i18n
+transifex-client==0.12.4
+edx-i18n-tools
+
+# for docker devstack
+docker-compose >= 1.5.1
```
Comment (mikix) [inline requirements/dev.in]: The versions in this file (which are added by you?) might better belong in contrainsts.txt -- ideally all the version constraints live there. But I get that for other files, you were just copying and pasting.

Comment (estute) [reply requirements/dev.in]: These are the versions pinned in `local.txt`. I just renamed it to `dev.in` to follow our naming convention in oep-18.

**Comment by estute** [comment]
I totally agree about removing the hard pinning, but since I don't have much context in this repo, I didn't want to break things.

**Comment by crice100** [comment]
@estute I think this repo is now OEP-18 compliant. Is there anything left to do for this PR?

**Comment by crice100** [comment]
Closing since I think this work has already been done. Feel free to re-open if I misunderstood or if there are still changes to be made for this.

**Pull request closed without merging**